react/components/ModalDialog/index.js

```diff
@@ -35,6 +35,7 @@ class ModalDialog extends Component {
         <div className="vtex-modal__confirmation flex justify-end mt8">
           <span className="mr4">
             <Button
+              testId={this.props.cancelation.testId}
```

Comment:
```suggestion
              testId={cancelation.testId}
```
You already have `const { confirmation, cancelation, loading, onClose } = this.props` in the render method.